### PR TITLE
Mesh export optimization for deformation motion blur

### PR DESCRIPTION
**Pull request by DMotorygin**
JIRA TICKET: RPRMAYA-2739

PURPOSE: Currently it takes too long for scene with deformation to export

EFFECT FOR THE USER: Faster scene export

### CODE REVIEW

File: FireRender.Maya.Src/Context/FireRenderContext.cpp [hunk 1]
```diff
@@ -2625,29 +2627,90 @@ bool FireRenderContext::Freshen(bool lock, std::function<bool()> cancelled)
...
 	}
 
+	const bool isDeformationMotionBlurEnabled = motionBlur() && TahoeContext::IsGivenContextRPR2(this) && !isInteractive();
+	const unsigned int motionSamplesCount = isDeformationMotionBlurEnabled ? motionSamples() : 1;
+
+	// read data from meshes
+	MTime initialTime = MAnimControl::currentTime();
+	MTime currentTime = initialTime;
+	unsigned int currentSampeIdx = 0;
+	//for (unsigned int currentSampeIdx = 0; currentSampeIdx < motionSamplesCount; ++currentSampeIdx)
```
Comment (Vsevolod1983) [inline FireRender.Maya.Src/Context/FireRenderContext.cpp:2672]: remove comment ?

File: FireRender.Maya.Src/Translators/MeshTranslator.cpp [hunk 2]
```diff
@@ -158,12 +178,42 @@ bool FireMaya::MeshTranslator::MeshPolygonData::Initialize(MFnMesh& fnMesh, unsi
...
+		assert(MStatus::kSuccess == mstatus);
+		std::copy(pData, pData + countNormals * 3, arrNormals.data());
+	}
+	else
 	{
-		motionSamplesCount = 0;
+		ReadDeformationFrame(fnMesh, 0);
 	}
 
+	//if (!ProcessDeformationFrameCount(fnMesh, fullDagPath))
```
Comment (Vsevolod1983) [inline FireRender.Maya.Src/Translators/MeshTranslator.cpp:204]: Remove ?

File: FireRender.Maya.Src/FireRenderObjects.cpp [hunk 3]
```diff
@@ -1113,7 +1113,7 @@ bool FireRenderMesh::setupDisplacement(std::vector<MObject>& shadingEngines, frw
 
 void FireRenderMesh::ReloadMesh(const MDagPath& meshPath)
 {
-	MMatrix mMtx = meshPath.inclusiveMatrix();
+	//MMatrix mMtx = meshPath.inclusiveMatrix();
```
Comment (Vsevolod1983) [inline FireRender.Maya.Src/FireRenderObjects.cpp:1116]: remove?

File: FireRender.Maya.Src/Context/FireRenderContext.cpp [hunk 4]
```diff
@@ -2654,29 +2656,90 @@ bool FireRenderContext::Freshen(bool lock, std::function<bool()> cancelled)
...
+
+		++currentSampeIdx;
+		if (currentSampeIdx == motionSamplesCount)
+			break;
+
+		// positioning on next point of time (starting from currentTime)
+		assert(currentSampeIdx < motionSamplesCount);
+		currentTime += (float)currentSampeIdx / (motionSamplesCount - 1);
+		MGlobal::viewFrame(currentTime);
+	} while (true);
```
Comment (Vsevolod1983) [inline FireRender.Maya.Src/Context/FireRenderContext.cpp:2728]: while true, is it really necessary ?

**Review by V-Gorash — APPROVED** [review]

**Review by Vsevolod1983 — APPROVED** [review]

**Pull request merged**